presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java

```diff
@@ -36,4 +39,18 @@ private MoreMaps() {}
                 .flatMap(Collection::stream)
                 .collect(toMap(Map.Entry::getKey, Map.Entry::getValue, merger));
     }
+
+    public static <K, V> Map<K, V> immutableLinkedMap(Map<K, V> map)
```

Comment:
Immutable collections preserve the creation order. Neither of these helpers are needed and should be replaced with the immutable versions.

"Except for sorted collections, order is preserved from construction time."

https://github.com/google/guava/wiki/ImmutableCollectionsExplained